src/app/main.rs

```diff
@@ -403,6 +404,58 @@ where
             config.destination_context.clone(),
         );
 
+        let trace_collector_svc = config.trace_collector_addr.as_ref().map(|addr| {
+            svc::stack(connect::svc())
+                .push(tls::client::layer(local_identity.clone()))
+                .push(keepalive::connect::layer(config.outbound_connect_keepalive))
+                .push_timeout(config.control_connect_timeout)
+                // TODO: perhaps rename from "control" to "grpc"
+                .push(control::client::layer())
+                .push(control::resolve::layer(dns_resolver.clone()))
+                // TODO: we should have metrics of some kind, but the standard
+                // HTTP metrics aren't useful for a client where we never read
+                // the response.
+                .push(reconnect::layer({
+                    let backoff = config.control_backoff.clone();
+                    move |_| Ok(backoff.stream())
+                }))
+                .push(proxy::grpc::req_body_as_payload::layer().per_make())
+                .push(control::add_origin::layer())
+                .push_buffer_pending(
+                    config.destination_buffer_capacity,
+                    config.control_dispatch_timeout,
+                )
+                .into_inner()
+                .make(addr.clone())
+        });
+
+        let (outbound_spans_tx, inbound_spans_tx) = match trace_collector_svc {
+            Some(trace_collector) => {
+                let (outbound_spans_tx, outbound_spans_rx) = mpsc::channel(100);
+                let (inbound_spans_tx, inbound_spans_rx) = mpsc::channel(100);
+
+                let node = oc::Node {
```

Comment:
Nit: again, it seems weird to me to do this much OpenCensus-specific setup in `main` --- I'd _prefer_ to see some of this in a function in a tracing module that we call with some configs.